### PR TITLE
feature/ORCA-464: Modify cumulus-orca deployment script to dynamically pull variables.

### PR DESCRIPTION
**Pull request by rizbihassan**
## Summary of Changes

Addresses [ORCA-464: Modify cumulus-orca deployment script to dynamically pull variables.](https://bugs.earthdata.nasa.gov/browse/ORCA-464)

## Changes

* updated bamboo spec to include deploying cumulus/orca modules into one stage
* updated release doc
* updated dockerfile for orca image to include jq
* deleted old scripts

## Type of change

Please delete options that are not relevant.

- [ x] New feature (non-breaking change which adds functionality)
- [ x] This change requires a documentation update

## How Has This Been Tested?

Ran new pipeline in bamboo

## Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [ x] Testing documentation
    - [ x] Development documentation
- [x ] My changes generate no new warnings

### CODE REVIEW

File: website/docs/developer/development-guide/code/versioning-releases.md [hunk 1]
```diff
@@ -156,8 +156,8 @@ Some of these buckets have cross-account IAM policies attached so that they can
 Hitting 'play' next to `Deploy DR ORCA buckets`, `Deploy Dev RDS Stack` and `Deploy Dev Cumulus and ORCA Stack` brings up a checkbox list to run multiple jobs at once. Note that none of the checkboxes should be checked.
 :::
 
-The Cumulus and TF buckets as well as dynamoDB table in cumulus OU account are created automatically in the `Deploy RDS cluster` and `Deploy Dev Cumulus and ORCA Stack` stages.
-These are the buckets that will be created in cumulus OU account:
+The Cumulus and TF buckets as well as dynamoDB table in cumulus OU account are created automatically in bamboo `Deploy ORCA Buckets and Cumulus/ORCA modules` stage. 
+These are the buckets that need to be created in cumulus OU account:
```
Comment (AndrewDorn) [inline website/docs/developer/development-guide/code/versioning-releases.md:160]: This was an intentional change on my part. "buckets need to be created" implies that the user must handle creation. "buckets will be created" states that they will be created without additional instruction.

File: website/docs/developer/development-guide/code/versioning-releases.md [hunk 2]
```diff
@@ -156,8 +156,8 @@ Some of these buckets have cross-account IAM policies attached so that they can
 Hitting 'play' next to `Deploy DR ORCA buckets`, `Deploy Dev RDS Stack` and `Deploy Dev Cumulus and ORCA Stack` brings up a checkbox list to run multiple jobs at once. Note that none of the checkboxes should be checked.
 :::
 
-The Cumulus and TF buckets as well as dynamoDB table in cumulus OU account are created automatically in the `Deploy RDS cluster` and `Deploy Dev Cumulus and ORCA Stack` stages.
-These are the buckets that will be created in cumulus OU account:
+The Cumulus and TF buckets as well as dynamoDB table in cumulus OU account are created automatically in bamboo `Deploy ORCA Buckets and Cumulus/ORCA modules` stage.
```
Comment (AndrewDorn) [inline website/docs/developer/development-guide/code/versioning-releases.md:159]: Incorrect grammar. If you want to add "Bamboo", it should read "are created automatically in the Bamboo `Deploy..."

File: website/docs/developer/development-guide/code/versioning-releases.md [hunk 3]
```diff
@@ -166,17 +166,17 @@ These are the buckets that will be created in cumulus OU account:
 - `<PREFIX>-protected`
 - `<PREFIX>-tf-state` (for storing the terraform state file in cumulus OU account)
 
-After hitting the play button on `Deploy Dev RDS Stack`, but before hitting `Run` in the popup, replace the following variables with yours.
+While running the `Deploy ORCA Buckets and Cumulus/ORCA modules` stage, replace the following variables with yours. This is because some variables are sensitive and some will vary depending upon the user running the pipeline. Hitting 'play' next to `Deploy DR ORCA buckets`, `Deploy ORCA Buckets and Cumulus/ORCA modules` brings up a checkbox list to run multiple jobs at once. Note that none of the checkboxes should be checked.
```
Comment (AndrewDorn) [inline website/docs/developer/development-guide/code/versioning-releases.md:169]: This was an intentional change on my part. This does not occur "while running", but before it is run.

File: website/docs/developer/development-guide/code/versioning-releases.md [hunk 4]
```diff
@@ -166,17 +166,17 @@ These are the buckets that will be created in cumulus OU account:
 - `<PREFIX>-protected`
 - `<PREFIX>-tf-state` (for storing the terraform state file in cumulus OU account)
 
-After hitting the play button on `Deploy Dev RDS Stack`, but before hitting `Run` in the popup, replace the following variables with yours.
+While running the `Deploy ORCA Buckets and Cumulus/ORCA modules` stage, replace the following variables with yours. This is because some variables are sensitive and some will vary depending upon the user running the pipeline. Hitting 'play' next to `Deploy DR ORCA buckets`, `Deploy ORCA Buckets and Cumulus/ORCA modules` brings up a checkbox list to run multiple jobs at once. Note that none of the checkboxes should be checked.
 
 - CUMULUS_AWS_ACCESS_KEY_ID
 - CUMULUS_AWS_SECRET_ACCESS_KEY
 - PREFIX
 - AWS_ACCOUNT_ID (for cumulus sandbox account)
```
Comment (AndrewDorn) [inline website/docs/developer/development-guide/code/versioning-releases.md:174]: Appears to now be unused.

**Review by AndrewDorn — CHANGES_REQUESTED** [review]
Minor documentation issues

File: website/docs/developer/development-guide/code/versioning-releases.md [hunk 5]
```diff
@@ -209,24 +209,6 @@ aws ec2 create-key-pair --key-name <PREFIX> --query 'KeyMaterial' --output text
...
-- DB_HOST_ENDPOINT
-- RDS_SECURITY_GROUP
-- RDS_USER_ACCESS_SECRET_ARN
-- CUMULUS_ORCA_DEPLOY_TEMPLATE_VERSION (optional)
-
-The RDS variables `RDS_SECURITY_GROUP`, `RDS_USER_ACCESS_SECRET_ARN` and `DB_HOST_ENDPOINT` can be found from output logs of the previous `Deploy Dev RDS Stack` stage as `security_group_id`, `admin_db_login_secret_arn`, and `rds_endpoint` respectively. Note that a new earthdata application will need to be first created if using a new prefix for new deployment which will give the values for `EARTHDATA_CLIENT_ID` and `EARTHDATA_CLIENT_PASSWORD`. `CUMULUS_ORCA_DEPLOY_TEMPLATE_VERSION` is the branch you want to check out in the [deployment repo](https://git.earthdata.nasa.gov/projects/ORCA/repos/cumulus-orca-deploy-template/browse) such as `v11.1.1-v4.0.1`.
+A new earthdata application will need to be first created if using a new prefix for new deployment which will give the values for `EARTHDATA_CLIENT_ID` and `EARTHDATA_CLIENT_PASSWORD`. `CUMULUS_ORCA_DEPLOY_TEMPLATE_VERSION` is the branch you want to check out in the [deployment repo](https://git.earthdata.nasa.gov/projects/ORCA/repos/cumulus-orca-deploy-template/browse) such as `v11.1.1-v4.0.1`.
```
Comment (bhazuka) [inline website/docs/developer/development-guide/code/versioning-releases.md:212]: We should only need to create the earthdata clientid and password once and then they can be used over and over again. The most you may have to do as an end user is update the redirect url for the application once this is setup. No need to create a new one each time you change a prefix. 
Screen Shot 2022-08-10 at 3 01 58 PM

In reality this is really only needed if you are calling the Cumulus API or doing things from the dashboard. In a majority of our integration tests so far, we are calling things directly from within the account so this setup may not be needed.

Definitely worth some discussion though as a team.

**Review by bhazuka — CHANGES_REQUESTED** [review]
A few minor tweeks and some things to discuss.

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 4 above)
Comment (bhazuka) [reply website/docs/developer/development-guide/code/versioning-releases.md:174]: I concur

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 3 above)
Comment (bhazuka) [reply website/docs/developer/development-guide/code/versioning-releases.md:169]: I concur

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 2 above)
Comment (bhazuka) [reply website/docs/developer/development-guide/code/versioning-releases.md:159]: I concur

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 1 above)
Comment (bhazuka) [reply website/docs/developer/development-guide/code/versioning-releases.md:160]: I concur

File: integration_test/deployment-orca.sh [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+  -input=false
+
+# Deploy cumulus-tf via terraform
+echo "Deploying Cumulus tf module to $bamboo_DEPLOYMENT"
+terraform apply \
+  -auto-approve \
+  -lock=false \
+  -input=false \
+  -var-file="terraform.tfvars" \
+  -var "cmr_username=$bamboo_CMR_USERNAME" \
```
[the remark this answers is not in the document]
Comment (rizbihassan) [reply integration_test/deployment-orca.sh:139]: This is being used here https://git.earthdata.nasa.gov/projects/ORCA/repos/cumulus-orca-deploy-template/browse/cumulus-tf/terraform.tfvars.example?at=release%2Fv11.1.1-v4.0.1#55

I wasn't sure if user would want to override the values in the future that's why wanted to keep these as variables here.

Are you suggesting these can be removed?

File: website/docs/developer/development-guide/code/versioning-releases.md [hunk 7]
```diff
@@ -166,17 +166,17 @@ These are the buckets that will be created in cumulus OU account:
...
 - CUMULUS_AWS_ACCESS_KEY_ID
 - CUMULUS_AWS_SECRET_ACCESS_KEY
 - PREFIX
 - AWS_ACCOUNT_ID (for cumulus sandbox account)
 - DB_ADMIN_PASSWORD
 - DB_USER_PASSWORD
-- AWS_SUBNET_ID1
-- AWS_SUBNET_ID2
-- VPC_ID
+- EARTHDATA_CLIENT_ID
```
[the remark this answers is not in the document]
Comment (rizbihassan) [reply website/docs/developer/development-guide/code/versioning-releases.md:177]: but EARTHDATA_CLIENT_ID can be different for different users unless everyone is using just one common application.

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 5 above)
Comment (rizbihassan) [reply website/docs/developer/development-guide/code/versioning-releases.md:212]: I can update the wording to say users don't have to create a new application if they already have one.

File: integration_test/deployment-orca.sh [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+mv terraform.tfvars.example terraform.tfvars
+
+#replacing terraform.tf with proper values
+sed -e 's/PREFIX/'"$bamboo_PREFIX"'/g; s/us-east-1/'"$bamboo_AWS_DEFAULT_REGION"'/g' terraform.tf.example > terraform.tf
+
+export AWS_ACCOUNT_ID=$(aws sts get-caller-identity | jq -r '.Account')
+export VPC_ID=$(aws ec2 describe-vpcs | jq -r '.Vpcs | to_entries | .[] | .value.VpcId')
+export AWS_SUBNET_ID1=$(aws ec2 describe-subnets --filters "Name=availability-zone,Values=us-west-2a"| jq -r '.Subnets | .[] | select (.Tags | .[] | .Value | contains ("Private application ")) | .SubnetId ')
+export AWS_SUBNET_ID2=$(aws ec2 describe-subnets --filters "Name=availability-zone,Values=us-west-2b"| jq -r '.Subnets | .[] | select (.Tags | .[] | .Value | contains ("Private application ")) | .SubnetId ')
+
```
[the remark this answers is not in the document]
Comment (rizbihassan) [reply integration_test/deployment-orca.sh:67]: will add in the 3rd subnet

File: integration_test/deployment-orca.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (rizbihassan) [reply integration_test/deployment-orca.sh:139]: will remove the 5 cmr variables

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 5 above)
Comment (rizbihassan) [reply website/docs/developer/development-guide/code/versioning-releases.md:212]: will keep the earthdata client ID

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 1 above)
Comment (rizbihassan) [reply website/docs/developer/development-guide/code/versioning-releases.md]: fixed

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 2 above)
Comment (rizbihassan) [reply website/docs/developer/development-guide/code/versioning-releases.md]: fixed

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 3 above)
Comment (rizbihassan) [reply website/docs/developer/development-guide/code/versioning-releases.md]: fixed

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 4 above)
Comment (rizbihassan) [reply website/docs/developer/development-guide/code/versioning-releases.md]: yes removed

**Review by AndrewDorn — APPROVED** [review]

File: website/docs/developer/development-guide/code/versioning-releases.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (bhazuka) [reply website/docs/developer/development-guide/code/versioning-releases.md:177]: good point.

**Review by bhazuka — APPROVED** [review]

**Pull request merged**